presto-hive/src/test/java/io/prestosql/plugin/hive/TestBackgroundHiveSplitLoader.java

```diff
@@ -358,8 +357,10 @@ public void testNoHangIfPartitionIsOffline()
         HiveSplitSource hiveSplitSource = hiveSplitSource(backgroundHiveSplitLoader);
         backgroundHiveSplitLoader.start(hiveSplitSource);
 
-        assertThrows(RuntimeException.class, () -> drain(hiveSplitSource));
-        assertThrows(RuntimeException.class, hiveSplitSource::isFinished);
+        assertThatThrownBy(() -> drain(hiveSplitSource))
+                .isInstanceOf(RuntimeException.class);
```

Comment:
can we be a bit more specific about exception type?
for example, we should not be OK (test should fail) when the code throws a NullPointerException
here it throws `PrestoException` and we should test for that

similarly, it is desireable to test the exception message
for example no message or some useless message won't be useful for end users.

thus, the asserion here should look like
```
assertThatThrownBy(() -> drain(hiveSplitSource))
        .isInstanceOf(PrestoException.class)
        .hasMessage("OFFLINE");
```

(i know that "OFFLINE" message is far from ideal, but let's keep fixing this as out of scope for the PR)

Reply:
Got it, pushed updates.

Reply:
Thanks! Would you be able to also add `.hasMessage` / `.hasMessageMatching` in places where we do not check the message at all yet?

Reply:
Updated